src/main/java/org/elasticsearch/index/translog/TranslogSnapshot.java

```diff
@@ -42,21 +42,19 @@
      * Create a snapshot of translog file channel. The length parameter should be consistent with totalOperations and point
      * at the end of the last operation in this snapshot.
      */
-    public TranslogSnapshot(List<ChannelSnapshot> orderedTranslogs, ESLogger logger) {
+    public TranslogSnapshot(List<Translog.Snapshot> orderedTranslogs) {
         this.orderedTranslogs = orderedTranslogs;
-        this.logger = logger;
         int ops = 0;
-        for (ChannelSnapshot translog : orderedTranslogs) {
+        for (Translog.Snapshot translog : orderedTranslogs) {
 
             final int tops = translog.estimatedTotalOperations();
             if (tops < 0) {
```

Comment:
the problem is that we always have to deal with the case where the replay didn't succeed we can't really get around this.